eval/run.rkt

```diff
@@ -56,13 +58,15 @@
     (define out
       (match hint
         [#t ; instruction should be reevaluated
-         (define start (current-inexact-milliseconds))
+         (define start-time (current-inexact-milliseconds))
+         (define start-memory (current-memory-use #f))
```

Comment:
Ah, I think you need to use cumulative here instead of #f for the argument. We want total allocated, even if there's been a GC in between, not just currently in use